base/strings/basic.jl

```diff
@@ -715,6 +717,7 @@ size(s::CodeUnits) = (length(s),)
 elsize(s::CodeUnits{T}) where {T} = sizeof(T)
 @propagate_inbounds getindex(s::CodeUnits, i::Int) = codeunit(s.s, i)
 IndexStyle(::Type{<:CodeUnits}) = IndexLinear()
+IteratorSize(::Type{<:CodeUnits}) = SizeUnknown()
```

Comment:
CodeUnits is a vector, not a string, and its length should always be known in O(1).